app/models.py

```diff
@@ -750,32 +754,69 @@ def pending_submission_reminder(cls, session: Session) -> "Query[Self]":
             col(cls.id).notin_(Message.application_by_type(MessageType.BORROWER_PENDING_SUBMIT_REMINDER)),
         )
 
+    @classmethod
+    def pending_external_onboarding_reminder(cls, session: Session) -> "Query[Self]":
+        """
+        Return query for SUBMITTED applications to lenders with external onboarding systems and whose borrower hasn't
+        already received a reminder to start the external onboarding process.
+
+        .. seealso:: :typer:`python-m-app-send-reminders`
+        """
+        return (
+            session.query(cls)
+            .filter(
+                cls.status == ApplicationStatus.SUBMITTED,
+                col(Lender.external_onboarding_url) != "",
+                col(cls.borrower_accessed_external_onboarding_at).is_(None),
+                col(cls.id).notin_(Message.application_by_type(MessageType.BORROWER_EXTERNAL_ONBOARDING_REMINDER)),
+            )
+            .join(Lender, cls.lender_id == Lender.id)
+        )
+
     @classmethod
     def lapseable(cls, session: Session) -> "Query[Self]":
         """
         Return a query for :meth:`~app.models.Application.unarchived` applications that have been waiting for the
-        borrower to respond (PENDING, ACCEPTED, INFORMATION_REQUESTED) for
-        :attr:`~app.settings.Settings.days_to_change_to_lapsed` days.
+        borrower to respond (PENDING, ACCEPTED, SUBMITTED (to a lender with external_onboarding_url set)
+        INFORMATION_REQUESTED) for :attr:`~app.settings.Settings.days_to_change_to_lapsed` days.
 
         .. seealso:: :typer:`python-m-app-update-applications-to-lapsed`
         """
         delta = timedelta(days=app_settings.days_to_change_to_lapsed)
 
-        return cls.unarchived(session).filter(
-            or_(
-                and_(
-                    cls.status == ApplicationStatus.PENDING,
-                    col(cls.created_at) + delta < datetime.now(),
-                ),
-                and_(
-                    cls.status == ApplicationStatus.ACCEPTED,
-                    col(cls.borrower_accepted_at) + delta < datetime.now(),
+        return (
+            cls.unarchived(session)
+            .filter(
+                or_(
+                    and_(
+                        cls.status == ApplicationStatus.PENDING,
+                        col(cls.created_at) + delta < datetime.now(),
+                    ),
+                    and_(
+                        cls.status == ApplicationStatus.ACCEPTED,
+                        col(cls.borrower_accepted_at) + delta < datetime.now(),
+                    ),
+                    and_(
+                        cls.status == ApplicationStatus.SUBMITTED,
+                        col(Message.created_at) + delta < datetime.now(),
```

Comment:
Why not compare to borrower_submitted_at? We don't care about the precise timing of messages elsewhere.

If we're planning on manually sending out external onboarding messages (to those who submitted before the external onboarding feature was added), we can either:

- Update their borrower_submitted_at to now() (my preferred option).
- Comment out and disable in Salt the lapsed command for 30 days.

Reply:
> If we're planning on manually sending out external onboarding messages

We will probably run this command manually the first time indeed, but even if we don't do that, the first time the cron job runs it, it will lapse all the applications, unless we update their `borrower_submitted_at` as you suggest. But why do you prefer that over checking the actual reminder submission date? I don't like losing the original submission date information, as we use that to compare with Clara's onboarding date etc